lib/imagefs.coffee

```diff
@@ -280,3 +327,13 @@ exports.listDirectory = (definition) ->
 			listDirectory(disk, definition.partition, definition.path)
 	else if definition.image instanceof filedisk.Disk
 		listDirectory(definition.image, definition.partition, definition.path)
+
+###*
+# @summary Closes the allocated resources. Call this when you're done using imagefs if you expect the program to end.
+# @function
+# @public
+#
+# @returns {Promise}
+###
+exports.close = ->
+	ext2fs.close()
```

Comment:
I think you're mixing up between the `close` method of the returned `fs` objects and the `imagefs.close` method.
Nothing in `resin-image-fs` is calling `ext2fs.close` by itself. The user has to explicitly call it himself via `imagefs.close`.

It doesn't close a disk image, it closes the resources allocated by ext2fs on import.

Unfortunately, the `beforeExit` event will never be triggered if you don't call `ext2fs.close` first.

I don't have a better solution for this, and if it is fixable, I think it should be fixed in `ext2fs`.